_episodes/38.md

```diff
@@ -0,0 +1,155 @@
+---
+layout: episode
+title:  "38: Expanding the Trino toolkit with polymorphic table functions"
+date: 2022-07-21
+tags: trino polymorphic table functions
+youtube_id: ""
+wistia_id: ""
+sections:
+- title: "Releases 387 to 390"
+  desc: "Details about features and fixes in the new releases"
+  time:
+---
+
+## Guests
+
+In this episode we have the pleasure to chat with a couple familiar faces who
+have been hard at work building and understanding the features we're talking
+about today:
+
+* [Kasia Findeisen](https://github.com/kasiafi), Trino Maintainer
+* [Martin Traverso](https://twitter.com/mtraverso), Trino Co-Creator & Maintainer
+
+## Releases 387 to 390
+
+Official highlights from Martin:
+
+More detailed information is available in the release notes for
+[Trino 387](https://trino.io/docs/current/release/release-387.html),
+[Trino 388](https://trino.io/docs/current/release/release-388.html),
+[Trino 389](https://trino.io/docs/current/release/release-389.html),
+and [Trino 390](https://trino.io/docs/current/release/release-390.html).
+
+## Concept of the episode: Polymorphic table functions
+
+### What is a table function?
+
+### What makes a table function polymorphic?
+
+## Pull request of the episode: PR 12325: Support query pass-through for JDBC-based connectors
+
+## Demos of the episode: The many ways you can leverage PTFs
+
+### Example 1: Using connector-specific features with query pass-through
+
+Trino always tries to implement the ANSI specification for SQL, and is pretty
+much committed to being compliant with it. However, Trino hasn't yet
+implemented every last function and clause from the ANSI SQL specification, so
+some databases may have features that Trino is yet to get to, and if you want
+to use those features, now you can, no need to wait for Trino to get there!
```

Comment:
I wouldn't focus on that so much. Trino is ahead of many databases in standard SQL support. With query pass-through, we will typically address the case when the underlying database has some native non-standard syntax, which cannot be rewritten to standard SQL, or which can be rewritten to standard SQL, but a PTF is used  as a contemporary enabler if someone wants to start using Trino right away, and take more time to adjust their queries. Another use case would be pushdown for performance.